libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -708,15 +708,16 @@ private void refreshAccessToken() throws IOException {
             // Then let's try to get a new authToken
             if (authTokenProvider != null) {
                 final String newAuthToken = authTokenProvider.getNewAuthToken();
-                if (newAuthToken != null) {
-                    setAuthToken(newAuthToken);
+
+				if (newAuthToken == null || authTokenProvider.getInstanceUrl() == null) {
+                    throw new RefreshTokenRevokedException("Could not refresh token");
                 }
 
+                // Use new token
+                setAuthToken(newAuthToken);
+
                 // Check if the instanceUrl changed
                 String instanceUrl = authTokenProvider.getInstanceUrl();
-                if (instanceUrl == null) {
```

Comment:
After a revoke, authTokenProvider.getInstanceUrl() would be null, so the old code would throw a IOException "Instance URL is null".
Now it throws a RefreshTokenRevokedException (which is a subclass of IOException - so backward compatible).